ceph/ceph_admin/add.py

```diff
@@ -34,10 +34,11 @@ def add(self: DaemonProtocol, config: Dict):
         if config.get("base_cmd_args"):
             base_cmd.append(config_dict_to_string(config["base_cmd_args"]))
 
-        pos_args = config["pos_args"]
-        node = pos_args[0]
-        host_id = get_node_by_id(self.cluster, node)
-        host = host_id.shortname
+        if config.get("pos_args"):
+            pos_args = config["pos_args"]
+            node = pos_args[0]
+            host_id = get_node_by_id(self.cluster, node)
+            host = host_id.shortname
```

Comment:
This changes the behavior. We need it to raise an issue when `pos_args` is not provided. With the new change, no `AttributeError` would be raised.

Reply:
I have actually fixed this in second commit, but when I did rebase somehow hose changes were disappeared.
Anyways, I provided a workaround for that.